app/controllers/posts_controller.rb

```diff
@@ -1,5 +1,20 @@
+require 'yaml'
 class PostsController < ApplicationController
-  def index; end
+  def index
+    @user = User.find(id)
+    @button_text = 'Pagination'
+    @path = helpers.user_post_path(id:)

```

Comment:
![image](https://user-images.githubusercontent.com/65068771/166853117-ae8dd504-bf21-4837-b3f8-942e502e40b9.png)

- [ ] kindly you not that you still have a syntax error here. Kindly make sure you solve it. The `user_id` you are trying to access is in the params, as you added a method for getting it(`def id`) you need to call it as a method(`id`).
- [ ] Note that also this path is incorrect, it should be `user_posts_path`

  Hint :point_down: 

```rb
@path = helpers.user_posts_path(id)
```